web_widget_timepicker/static/src/js/web_widget_timepicker.js

```diff
@@ -0,0 +1,115 @@
+odoo.define('web_widget_timepicker.form_widgets', function (require) {
+    "use strict";
+
+    var core = require('web.core');
+    var formats = require('web.formats');
+    var common = require('web.form_common');    
+
+    // Snippet from http://stackoverflow.com/questions/9036429/convert-object-string-to-json
+    function cleanup_str2json(str) {
+      return (str.length > 0 && str !== undefined) ? str
+        // wrap keys without quote with valid double quote
+        .replace(/([\$\w]+)\s*:/g, function(_, $1){return '"'+$1+'":'})    
+        // replacing single quote wrapped ones to double quote 
+        .replace(/'([^']+)'/g, function(_, $1){return '"'+$1+'"'}) : undefined;         
+    };
+
+    var TimePicker = common.AbstractField.extend(common.ReinitializeFieldMixin, {
+        is_field_number: true,        
+        template: "TimePickerField",
+        internal_format: 'float_time',
+        widget_class: 'oe_form_field_time',
+        events: {
+            'change input': 'store_dom_value',
+        },
+        init: function (field_manager, node) {
+            this._super(field_manager, node);
+            
+            this.internal_set_value(0);
+
+            this.options = _.defaults( {}, {
+                disableTextInput: true,
+                forceRoundTime: true,
+                step: 15,
+                selectOnBlur: true,
+                timeFormat: 'H:i',
+                scrollDefault: 'now',
+            });
+        },
+        initialize_content: function() {
+            if(!this.get("effective_readonly")) {

```

Comment:
why do you jump through all this hoops when the framework parses the `options` attribute (without `data-`) for you as json already? Compare with for example https://github.com/OCA/OCB/blob/9.0/addons/web/static/src/js/widgets/progress_bar.js#L31